File: src/main/java/com/rwbymod/block/BlockRegistry.java
package com.rwbymod.block;

import com.rwbymod.item.ItemBlockMeta;

import net.minecraft.block.Block;
import net.minecraft.item.ItemBlock;
import net.minecraftforge.fml.common.registry.GameRegistry;

public class BlockRegistry {
	
	public static Block dustOre;

	public static void init() {
		dustOre = register(new BlockDustOre(), true);
	}
	
	// ----- UTILITY FUNCTIONS ----- //
	
	private static Block register(Block block, boolean hasMeta) {
		GameRegistry.register(block);
		GameRegistry.register(hasMeta ? new ItemBlockMeta(block) : new ItemBlock(block), block.getRegistryName());
		return block;
	}
	private static Block register(Block block) { return register(block, false); }
	
}

File: src/main/scala/com/rwbymod/registry/BlockRegistry.scala
package com.rwbymod.registry

import net.minecraft.block.Block
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraft.item.ItemBlock
import com.rwbymod.block.BlockDustOre
import com.rwbymod.item.ItemBlockMeta

object BlockRegistry {

  val dustOre = new BlockDustOre()

  def init {
    //Non-meta blocks
    Array[Block](
    ).foreach { block => {
      GameRegistry.register(block)
      GameRegistry.register(new ItemBlock(block), block.getRegistryName)
    }}

    //Meta blocks
    Array[Block](
      dustOre
    ).foreach { block => {
      GameRegistry.register(block)
      GameRegistry.register(new ItemBlockMeta(block), block.getRegistryName)
    }}
  }

}